omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java

```diff
@@ -66,6 +72,37 @@ public PatientQueue getByUniqueId(String uniqueId) {
 		return patientQueue;
 	}
 	
+	@Override
+	public Object update(String uuid, SimpleObject propertiesToUpdate, RequestContext context) throws ResponseException {
+		if (propertiesToUpdate.get("status") == null) {
+			return super.update(uuid, propertiesToUpdate, context);
+		}
+		PatientQueue patientQueue = getPatientQueueForUpdate(uuid, propertiesToUpdate);
+		ValidateUtil.validate(patientQueue);
+		patientQueue.setDateChanged(new Date());
+
+		if (propertiesToUpdate.get("status") != null && propertiesToUpdate.get("status") != "picked") {
+			patientQueue.setDatePicked(new Date());
+		} else if (propertiesToUpdate.get("status") != null && propertiesToUpdate.get("status") != "completed") {
+			patientQueue.setDateCompleted(new Date());
```

Comment:
Why do you set date completed when status is not completed?

Reply:
@dkayiwa done responding to comment